src/plugins/walletConnectToDapps/WalletConnectBridgeProvider.tsx

```diff
@@ -0,0 +1,500 @@
+            })()
+          } catch (error) {
+            moduleLogger.error(
+              error,
+              { fn: 'approveRequest:eth_sendTransaction' },
+              'Error sending transaction',
+            )
+          }
+          break
+        }
+        case 'eth_signTransaction': {
+          const tx = request.params[0]
+          const advancedParams = approveData as ETHSignTx
+          const addressNList = toAddressNList(accountMetadataById[wcAccountId].bip44Params)
+          const nonce = advancedParams.nonce ? convertNumberToHex(advancedParams.nonce) : tx.nonce
+          const gasLimit = advancedParams.gasLimit
+            ? convertNumberToHex(advancedParams.gasLimit)
+            : tx.gas
+          result = await chainAdapter.signTransaction({
+            txToSign: {
+              addressNList,
+              chainId: parseInt(fromAccountId(wcAccountId).chainReference),
+              data: tx.data,
+              gasLimit,
+              nonce,
+              to: tx.to,
+              value: tx.value,
+            },
+            wallet,
+          })
+          break
+        }
+        default:
+          break
+      }
+      if (result) {
+        connector.approveRequest({ id: request.id, result })
+        setCallRequest(undefined)
+      } else {
+        const message = 'JSON RPC method not supported'
+        connector.rejectRequest({ id: request.id, error: { message } })
+        setCallRequest(undefined)
+      }
+    },
+    [wallet, wcAccountId, connector, signMessage, signTypedData, accountMetadataById],
+  )
+
+  const rejectRequest = useCallback(
+    (request: WalletConnectCallRequest) => {
+      const payload = { error: { message: 'User rejected request' }, id: request.id }
+      connector?.rejectRequest(payload)
+      setCallRequest(undefined)
+    },
+    [connector],
+  )
+
+  // const handleSessionUpdate = useCallback(
+  //   (...args: any) => {
+  //     if (!connector) return
+  //     if (!wcAccountId) return
+  //     const { chainId, account: address } = fromAccountId(wcAccountId)
+  //     const chainAdapter = getChainAdapterManager().get(chainId)
+  //     if (!chainAdapter) return
+  //     // connector.updateSession({
+  //     //   chainId: parseInt(fromChainId(chainId).chainReference),
+  //     //   accounts: [address],
+  //     // })
+  //   },
+  //   [connector, wcAccountId],
+  // )
+
+  const handleDisconnect = useCallback(() => {
+    connector?.off('session_request')
+    connector?.off('session_update')
+    connector?.off('connect')
+    connector?.off('disconnect')
+    connector?.off('call_request')
+    try {
+      connector?.killSession()
+    } catch (e) {
+      moduleLogger.error(e, { fn: 'handleDisconnect' }, 'Error killing session')
+    }
+    setDapp(null)
+    setConnector(undefined)
+    localStorage.removeItem('walletconnect')
+  }, [connector])
+
+  // if evmChainId or wallet changes, update the walletconnect session
+  useEffect(() => {
```

Comment:
this needs to be combined with the wallet effect below, they're both doing the same thing